### PR TITLE
binutils: don't build with glibc@2.13

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by danielnachun** [comment]
Now that the `binutils` formula isn't used to bootstrap `glibc`, we can remove all the workarounds we used to make that bootstrapping work.  I went straight to disabling the `gold` formula because we can just build it as part of `binutils` as we did before.  If it should instead be deprecated or deleted, I'm fine either way.  Regardless it doesn't make sense to maintain as a separate formula.

**Comment by Bo98** [comment]
Our deprecation policy does not allow the disabling of formula without a 3 month deprecation period.

We however do have strategies for merging formulae, such as `formula_renames.json`.

**Comment by danielnachun** [comment]
I think we can use `formula_renames.json` here.  Does that mean I add `gold` to there and delete the `gold` formula?

**Comment by Bo98** [comment]
I think so

**Comment by danielnachun** [comment]
I changed this so that `gold` is added to `formula_renames.json`.  Now if a user tries to install `gold`, they'll just get redirected to `binutils` instead which provides `gold`.

File: Formula/binutils.rb [hunk 1]
```diff
@@ -36,27 +34,25 @@ def install
...
 
     if OS.mac?
       Dir["#{bin}/*"].each do |f|
         bin.install_symlink f => "g" + File.basename(f)
       end
     else
       # Reduce the size of the bottle.
-      system "strip", *bin.children, *lib.glob("*.a")
+      bin_files = bin.children
+      bin_files -= [bin/"gp-display-html"] # This is not a binary executable.
```
Comment (carlocab) [inline Formula/binutils.rb:54]:
```suggestion
      bin_files = bin.children.select(&:elf?)
```
Something like this should work.

**Comment by carlocab** [comment]
You probably also need to do `link_overwrite` here to avoid `brew link` conflicts for anyone who's done `brew install gold` in the meantime.

**Comment by danielnachun** [comment]
> You probably also need to do `link_overwrite` here to avoid `brew link` conflicts for anyone who's done `brew install gold` in the meantime.

Good catch, I added `link_overwrite` lines for the 3 files affected by this.

**Comment by danielnachun** [comment]
I've temporarily remove `long build` and `CI-linux-self-hosted` so the test can run just to make sure `binutils` itself passes.  Once that is, `long build` and `CI-linux-self-hosted` can be added back and we can do a full dependent testing run.

File: Formula/binutils.rb (hunk 1 above)
Comment (danielnachun) [reply Formula/binutils.rb]: It worked great!  That's a really neat solution if we run into this problem again.

**Comment by danielnachun** [comment]
`binutils` itself built fine after the additional changes.  I am going to cancel this now and the `long build` and `CI-linux-self-hosted` labels back so we can do full dependent testing run.

**Comment by danielnachun** [comment]
Actually this doesn't need `long build` because the macOS build will finish in less than 90 minutes.  It just needs the self-hosted runner.

**Comment by danielnachun** [comment]
This is very weird but the tests are failing because somehow brewed `glibc` got installed.  I'm cancelling for now while we figure out what happened.

**Comment by danielnachun** [comment]
The Linux CI run now passes with no issues. This will be ready to merge when the macOS CI runs are done.

**Review by carlocab — DISMISSED** [review]

**Pull request closed without merging**